File: domain/src/main/java/org/openmedstack/eventstore/EventMessage.java
package org.openmedstack.eventstore;

import java.util.HashMap;

public final class EventMessage {
    private Object _body;
    private HashMap<String, Object> _headers;

    public EventMessage(Object body, HashMap<String, Object> headers) {
        _headers = headers == null ? new HashMap<>() : headers;
        _body = body;
    }

    public HashMap<String, Object> getHeaders() {
        return _headers;
    }

    public Object getBody() {
        return _body;
    }
}

File: domain/src/main/kotlin/org/openmedstack/eventstore/EventMessage.kt
package org.openmedstack.eventstore

class EventMessage(body: Any, headers: HashMap<String, Any>) {
    private val _body: Any
    private val _headers: HashMap<String, Any>
    val headers: HashMap<String, Any>
        get() = _headers

    val body: Any
        get() = _body

    init {
        _headers = headers
        _body = body
    }
}